app/models/manageiq/providers/azure/manager_mixin.rb

```diff
@@ -50,7 +50,7 @@ def connection_rescue_block
       raise # Raise before falling into catch-all block below
     rescue StandardError => err
       _log.error("Error Class=#{err.class.name}, Message=#{err.message}, Backtrace=#{err.backtrace}")
-      raise MiqException::MiqInvalidCredentialsError, _("Unexpected response returned from system: %{error_message}") % {:error_message => err.message}
+      raise MiqException::MiqUnreachableError, _("Unexpected response returned from system: %{error_message}") % {:error_message => err.message}
```

Comment:
Perhaps it makes sense to just re-raise err because we're not really sure it's unreachable at this point.  We treat `StandardError` exceptions the same as `MiqUnreachableError`: we retry more frequently with an exponential backoff.  The problem was that we were explicitly raising `MiqInvalidCredentialsError`, an exception we don't retry as frequently.

https://github.com/ManageIQ/manageiq/blob/d101c7997d556f08a7b0747703654ba74915b7da/app/models/mixins/authentication_mixin.rb#L401-L409

Reply:
@jrafanie Fair enough, I'll change it.